jupyter-book/introduction/raw_data_processing.md

```diff
@@ -1,52 +1,397 @@
+where the steps described in this section occur and to convey what different processing options are possible. These commands will be run from the command line, and [`conda`](https://docs.conda.io/en/latest/) will be used for installing all of the software required for running this example.
+
+(raw-proc:example-prep)=
+### Preparation
+
+Before we start, we create a conda environment in the terminal and install the required packages. In this procedure, we will install [`alevin-fry`](https://alevin-fry.readthedocs.io/en/latest/), [`salmon`](https://salmon.readthedocs.io/en/latest/) and [`pyroe`](https://github.com/COMBINE-lab/pyroe).
+
+```bash
+conda create -n af_xmpl -y -c bioconda python=3.9 salmon alevin-fry pyroe
+conda activate af_xmpl
+```
+
+```{admonition} Note on using an M1/M2-based device
+
+Conda does not currently build most packages natively for Apple silicon. Therefore, if you 
+are using a non-Intel-based Apple computer (e.g., with an M1(Pro/Max/Ultra) or M2 chip), you 
+should make sure to specify that your environment uses the Rosetta2 translation layer.  
+To do this, you can replace the above commands with the following (instructions adopted 
+from [here](https://github.com/Haydnspass/miniforge#rosetta-on-mac-with-apple-silicon-hardware)):
+
+```bash
+CONDA_SUBDIR=osx-64 conda create -n af_xmpl -y -c bioconda python=3.9 salmon alevin-fry pyroe   # create a new environment
+conda activate af_xmpl
+conda env config vars set CONDA_SUBDIR=osx-64  # subsequent commands use intel packages
+```
+
+Next, we create a working directory `af_xmpl_run`, and download and uncompress the example dataset from a remote host.
+
+```bash
+# create a working dir 
+mkdir af_xmpl_run && cd af_xmpl_run
+
+# download the example dataset
+wget -qO- https://umd.box.com/shared/static/lx2xownlrhz3us8496tyu9c4dgade814.gz | tar xzf - --strip-components=1 -C .
+
+# download CB permit list 
+wget https://raw.githubusercontent.com/10XGenomics/cellranger/master/lib/python/cellranger/barcodes/3M-february-2018.txt.gz && gunzip 3M-february-2018.txt.gz
+
+```
+
+(raw-proc:example-map)=
+### Mapping the data
+
+Now, we process the data to obtain the mapping information. This includes three steps:
+1. Build the _splici_ reference (<u>splic</u>ed transcripts + <u>i</u>ntrons), using the genome and gene annotation file.
+2. Index the _splici_ reference.
+3. Map the reads against the _splici_ index.
+
+```bash
+# making splici reference
+# pyroe make-splici genome_file gtf_file read_length out_dir
```

Comment:
Where does the read length come from? (Expanding the comment might help)

Reply:
Done.